dartagnan/src/main/java/com/dat3m/dartagnan/wmm/Definition.java

```diff
@@ -0,0 +1,106 @@
+package com.dat3m.dartagnan.wmm;
+
+import com.dat3m.dartagnan.encoding.EncodingContext;
+import com.dat3m.dartagnan.program.filter.FilterAbstract;
+import com.dat3m.dartagnan.wmm.utils.Tuple;
+import org.sosy_lab.java_smt.api.BooleanFormula;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public abstract class Definition implements Constraint {
+
+    protected final Relation definedRelation;
+    final String term;
```

Comment:
What is this `term` variable used for?

Reply:
It is more a `termPattern` at this point.  It replaces overriding `toString()` in all specializations of `Definition`.